layer.go

```diff
@@ -75,9 +76,27 @@ type DriverInfo struct {
 
 type GUID [16]byte
 
+func guidToArray(g guid.GUID) GUID {
```

Comment:
I realized we have this `GUID [16]byte` type which is exported here. I think it's probably more trouble than it's worth to try to change all usage of this type to go-winio's `guid.GUID`, so I just made some helpers to convert to and from this type instead. Open to suggestions on something better to do here instead.

Reply:
I think this is fine. It may be interesting to add such functions to go-winio's GUID type to make it easier to convert between that and other GUIDs.

Reply:
Agreed, I'll work on a change to add them.